packages/api/src/Base.ts

```diff
@@ -589,37 +592,50 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
 
     decorated.creator = creator;
 
-    decorated.at = (hash: Hash | Uint8Array | string, arg?: CodecArg): C =>
+    decorated.at = (hash: Hash | Uint8Array | string, arg1?: CodecArg, arg2?: CodecArg): C =>
       onCall(
-        (arg: CodecArg) => this._rpcRx.state.getStorage([creator, arg], hash),
-        [arg]
+        (...args: CodecArg[]) => this._rpcRx.state.getStorage(
+          creator.meta.type.isDoubleMap
+            ? [creator, args]
+            : [creator, ...args],
+          hash),
+        [arg1, arg2]
       ) as C;
 
     // FIXME The unknown cast is needed since the onCall result, `C | S` cannot
     // be converted from C to the actual result required
-    decorated.hash = (arg?: CodecArg): HashResult<C, S> =>
+    decorated.hash = (arg1?: CodecArg, arg2?: CodecArg): HashResult<C, S> =>
       onCall(
-        (arg: CodecArg) => this._rpcRx.state.getStorageHash([creator, arg]),
-        [arg]
+        (...args: CodecArg[]) => this._rpcRx.state.getStorageHash(
+          creator.meta.type.isDoubleMap
+            ? [creator, args]
+            : [creator, ...args]),
+        [arg1, arg2]
       ) as unknown as HashResult<C, S>;
 
-    decorated.key = (arg?: CodecArg): string =>
-      u8aToHex(compactStripLength(creator(arg))[1]);
+    decorated.key = (arg1?: CodecArg, arg2?: CodecArg): string =>
+      creator.meta.type.isDoubleMap
+      ? u8aToHex(creator([arg1, arg2]))
+      : u8aToHex(compactStripLength(creator(arg1))[1]);
 
-    decorated.multi = (args: Array<CodecArg>, callback?: CodecCallback): S =>
+    // When using double map storage function, user need to path double map key as an array
+    decorated.multi = (args: Array<CodecArg[] | CodecArg>, callback?: CodecCallback): S =>
       onCall(
         () =>
           this._rpcRx.state
-            .subscribeStorage(args.map((arg) => [creator, arg]))
+            .subscribeStorage(args.map((arg: CodecArg[] | CodecArg) => [creator, arg]))
             .pipe(map((results) => new VectorAny(...results))),
         [],
         callback
       ) as unknown as S;
 
-    decorated.size = (arg?: CodecArg): U64Result<C, S> =>
+    decorated.size = (arg1?: CodecArg, arg2?: CodecArg): U64Result<C, S> =>
       onCall(
-        (arg: CodecArg) => this._rpcRx.state.getStorageSize([creator, arg]),
-        [arg]
+        (...args: CodecArg[]) => this._rpcRx.state.getStorageSize(
+          creator.meta.type.isDoubleMap
+            ? [creator, args]
+            : [creator, ...args]),
```

Comment:
My only concern here is that it effectively ends up being `[creator, <value of arg1>, undefined]` when called with 1 arg, instead of `[creator, <value of arg1>]`. Atm, no difference since the normal keys only accept one arg and would be split there, but just worried it could have some consequences down the line?

Reply:
what about this?
``` 
decorated.size = (arg1?: CodecArg, arg2?: CodecArg): U64Result<C, S> =>
      onCall(
        (arg1?: CodecArg, arg2?: CodecArg) => this._rpcRx.state.getStorageSize(
          creator.meta.type.isDoubleMap
            ? [creator, [arg1, arg2]]
            : [creator, arg1]),
        [arg1, arg2]
      ) as unknown as U64Result<C, S>;
```
Now when called with 1 arg, it will be `[creator, <value of arg1>]`

Reply:
It think that is fine since the second arg is anyway (currently) useless in any other non-DoubleMap operations.

Reply:
Cool. I updated `.at()`, `.hash()` and `.size()`. Also I rerun the test, and all tests passed.